xscen/catalog.py

```diff
@@ -347,6 +347,113 @@ def exists_in_cat(self, **columns):
             logger.info(f"An entry exists for: {columns}")
         return exists
 
+    def to_dataset(
+        self,
+        concat_on: Optional[Union[List[str], str]] = None,
+        ensemble_on: Optional[Union[List[str], str]] = None,
+        calendar: Optional[str] = "standard",
+        **kwargs,
+    ) -> xr.Dataset:
+        """
+        Open the catalog's entries into a single dataset.
+
+        Same as :py:meth:`~intake_esm.core.esm_datastore.to_dask`, but with additionnal control over the aggregations.
+        Ensemble preprocessing logic is taken from :py:func:`xclim.ensembles.create_ensemble`.
+        When `ensemble_on` is given, the function ensures all entries have the correct time coordinate according to `xrfreq`.
+
+        Parameters
+        ----------
+        concat_on : list of strings or str, optional
+          A list of catalog columns to concat the datasets over. Each will become a new dimension with the column values as coordinates.
```

Comment:
I thought I understood what `to_dataset` did until I read this line...

`additionnal control over the aggregations` sounded to me like we could change the aggregation controls, as is implied in the text and in the code, but as far as I know, this does not add dimensions to the dataset? It's more of a way to say `"These combination of columns define which files can be opened at the same time"`, and then you have to use `xarray_combine_by_coords_kwargs` if dimensions do not match.

Reply:
It does add dimension. For me (and maybe for you), the use of "aggregation" was confusing. Elsewhere in xscen, aggregation means average and reducing the time or space dimension. Here, it means adding new dimension to the dataset, similar to create_ensemble (aggregating many datasets in one).

Reply:
My bad, I confused `aggregation_control.aggregations` with `aggregation_control.groupby_attrs`. That being said, I think the name and description is kind of confusing. I'll think about a suggestion.